main.go

```diff
@@ -117,16 +117,26 @@ func handle(w http.ResponseWriter, r *http.Request) {
 
 	switch filepath.Base(fpath) {
 	case "index.html", ".":
+		htmlTemplate := indexHTML
 		if _, err := os.Stat(fpath); err != nil && !os.IsNotExist(err) {
 			http.Error(w, err.Error(), http.StatusInternalServerError)
 			return
+		} else if fp, err := os.Open("index.html"); err == nil {
```

Comment:
If `err != nil && errors.Is(err, fs.ErrNotExist)`, this error should be reported as `http.Error`

Reply:
If I'm understanding your suggestion correctly that condition would trigger if the user has not defined their own `index.html`, which I believe would break `wasmserve` as it is used today, no?

Reply:
Ah you are right...

Also, I start to think why the first `os.Stat` exist for index.html... From my investigation, I found there was a fallback in the initial state https://github.com/hajimehoshi/wasmserve/commit/0e21cc99ec1d8904a6fbdf17ff204cf1cf921196 but this disappeared accidentally with wrong early-returns. 

Let me revert this change.